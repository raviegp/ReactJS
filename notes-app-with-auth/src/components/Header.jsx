import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

import SearchBar from './SearchBar';
import { Button } from '@material-ui/core';
import DescriptionIcon from '@material-ui/icons/Description';
import { firebase, googleAuthProvider } from '../firebase/firebase';

const styles = theme => ({
  root: {
    width: '100%',
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  grow: {
    flexGrow: 1
  },
});

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: localStorage.getItem('isLoggedIn') || false,
    }
    this.handleLogin = this.handleLogin.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogin() {
    return firebase.auth().signInWithPopup(googleAuthProvider).then(success => {
      localStorage.setItem('isLoggedIn', true);
      this.setState({ isLoggedIn: true });
    })
  }

  handleLogout() {
    return firebase.auth().signOut().then(() => {
      localStorage.removeItem('isLoggedIn');
      this.setState({ isLoggedIn: false });
    }).catch(function (error) {
      // An error happened.
    });

  }

  render() {
    const { classes } = this.props;
    // Object destructruing -- if assignee and variable name is same then we can do below
    // instead of isLoggedIn = this.state.isLoggedIn -- as both we are using same name
    const { isLoggedIn } = this.state;
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <DescriptionIcon />
            <Typography className={classes.title} variant="h6" color="inherit" noWrap>
              Keep - Notes
            </Typography>
            {isLoggedIn ? <SearchBar filterNotes={this.props.filterNotes}/> : ''}
            <div className={classes.grow} />
            <Button onClick={isLoggedIn ? this.handleLogout : this.handleLogin}>
              {isLoggedIn ? 'Logout' : 'Login'}
            </Button>
          </Toolbar>
        </AppBar>
      </div>
    );
  }

}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);    // HOC -> Higher Order Component which takes another compoennt as an argument