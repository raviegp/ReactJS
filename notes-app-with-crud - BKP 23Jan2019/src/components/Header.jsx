import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import Button from '@material-ui/core/Button';
import SearchBar from './SearchBar';

const styles = theme => ({
  root: {
    width: '100%',
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  button: {
    margin: theme.spacing.unit,
  }
});

function Header(props) {
  const { classes } = props;

  function onToggleChange() {
    const changeView = props.viewType === 'gridView' ? 'listView' : 'gridView';
    props.toggleView(changeView);
  }

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton className={classes.menuButton} color="inherit" aria-label="Open drawer">
            <MenuIcon />
          </IconButton>
          <Typography className={classes.title} variant="h6" color="inherit" noWrap>
            Keep - Notes
          </Typography>
          <div>
            <Button variant="contained" href="#contained-buttons" className={classes.button} onClick={onToggleChange}>
              {props.viewType === 'gridView' ? 'List' : 'Grid'}
            </Button>
          </div>
          <SearchBar />
        </Toolbar>
      </AppBar>
    </div>
  );
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);    // HOC -> Higher Order Component which takes another compoennt as an argument
