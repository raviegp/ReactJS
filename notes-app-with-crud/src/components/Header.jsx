import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import SearchBar from './SearchBar';
import DescriptionIcon from '@material-ui/icons/Description';

const styles = theme => ({
  root: {
    width: '100%',
    alignItems: 'center'
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  button: {
    margin: theme.spacing.unit * 2,
    color: 'white',
    height: theme.spacing.unit * 3,
    padding: 0,
    marging: 0
  }
});

class Header extends Component {

  constructor(props) {
    super(props);
    this.onToggleChange = this.onToggleChange.bind(this);
  }

  onToggleChange() {
    const changeView = this.props.viewType === 'gridView' ? 'listView' : 'gridView';
    this.props.toggleView(changeView);
  }

  render() {
    const { classes, viewType } = this.props;

    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <DescriptionIcon />
            <Typography className={classes.title} variant="h6" color="inherit" noWrap>
              Keep - Notes
            </Typography>
            <Button variant="outlined" color="secondary" className={classes.button} onClick={this.onToggleChange}>
              {viewType === 'gridView' ? 'List' : 'Grid'}
            </Button>
            <SearchBar filterNotes={this.props.filterNotes} />
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);    // HOC -> Higher Order Component which takes another compoennt as an argument
