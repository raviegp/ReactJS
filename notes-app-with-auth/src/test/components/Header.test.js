import React from 'react';
import { mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import Header from '../../components/Header';
 
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

test('test Header page', () => {
    const wrapper = mount(<Header />);
    expect(wrapper).toMatchSnapshot();
});