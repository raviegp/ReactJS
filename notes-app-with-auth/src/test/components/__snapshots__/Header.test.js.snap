// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header page 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionStatic-9 MuiAppBar-colorPrimary-12"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-45"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
        />
      </svg>
      <h6
        className="MuiTypography-root-54 MuiTypography-h6-71 MuiTypography-colorInherit-83 MuiTypography-noWrap-80 Header-title-3"
      >
        Keep - Notes
      </h6>
      <div
        className="Header-grow-4"
      />
      <button
        className="MuiButtonBase-root-115 MuiButton-root-89 MuiButton-text-91 MuiButton-flat-94"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-90"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-118"
        />
      </button>
    </div>
  </header>
</div>
`;
